### PR TITLE
refactor: Convert ApiUtilsTest to Kotlin

### LINKED ISSUE #12130: [Bug] IDE showing errors in API project
###### Reproduction Steps

1.  Open ApiUtilsTest

###### Expected Result

It should now show any errors.

###### Actual Result

It is showing these errors:
image

###### Research

As suggested by @Arthur-Milchior [here](https://github.com/ankidroid/Anki-Android/issues/11917#issuecomment-1221608889).
We should remove the Utils file & replace any of their use with the standard method we have there.

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Note: I would love to work on this issue. Can anyone please assign this issue to me?**

### PR DESCRIPTION
**Pull request by david-allison**
com.ichi2.anki.api.ApiUtilsTest

Git hooks were disabled for this commit due to the extra commit for the
java -> kt rename

(irrelevant) Issue with conversion file was listed: #12209

added `kotlin-test` for null-safe: `assertNull` required for lint

Also closes #12130

### CODE REVIEW

File: api/src/test/java/com/ichi2/anki/api/ApiUtilsTest.kt [hunk 1]
```diff
@@ -1,81 +1,86 @@
...
-import java.util.Set;
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNull;
+import com.ichi2.anki.api.Utils.fieldChecksum
+import com.ichi2.anki.api.Utils.joinFields
+import com.ichi2.anki.api.Utils.joinTags
+import com.ichi2.anki.api.Utils.splitFields
+import com.ichi2.anki.api.Utils.splitTags
+import org.junit.Assert
```
Comment (lukstbit) [inline api/src/test/java/com/ichi2/anki/api/ApiUtilsTest.kt:10]: Import again Assert.assertEquals

File: api/src/test/java/com/ichi2/anki/api/ApiUtilsTest.kt [hunk 2]
```diff
@@ -1,81 +1,86 @@
...
  */
-
-@RunWith(RobolectricTestRunner.class)
-public class ApiUtilsTest {
-
-    private static final String delimiter = "\u001F";
-
+// TODO: Kotlin Cleanup
+@RunWith(RobolectricTestRunner::class)
+class ApiUtilsTest {
```
Comment (lukstbit) [inline api/src/test/java/com/ichi2/anki/api/ApiUtilsTest.kt:29]: @KotlinCleanup("replace Assert.assertEquals with kotlin.test.assertEquals")

File: api/src/test/java/com/ichi2/anki/api/ApiUtilsTest.kt [hunk 3]
```diff
@@ -1,81 +1,86 @@
...
 
     @Test
-    public void joinTagsShouldReturnEmptyStringWhenSetIsValid() {
-        Set<String> set = new HashSet<>();
-        set.add("A");
-        set.add("B");
-        set.add("C");
-        assertEquals("A B C", Utils.joinTags(set));
+    fun joinTagsShouldReturnEmptyStringWhenSetIsValid() {
+        val set: MutableSet<String?> = HashSet()
```
Comment (lukstbit) [inline api/src/test/java/com/ichi2/anki/api/ApiUtilsTest.kt:52]: @KotlinCleanup("use mutableSetOf, use scope function")

File: api/src/test/java/com/ichi2/anki/api/ApiUtilsTest.kt [hunk 4]
```diff
@@ -1,81 +1,86 @@
...
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.robolectric.RobolectricTestRunner;
-
-import java.util.HashSet;
-import java.util.Set;
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNull;
+import com.ichi2.anki.api.Utils.fieldChecksum
```
Comment (lukstbit) [inline api/src/test/java/com/ichi2/anki/api/ApiUtilsTest.kt:5]: Revert the imports of the individual methods from Utils.

**Review by lukstbit — CHANGES_REQUESTED** [review]

**Comment by david-allison** [comment]
Done. Using `TODO` as there's no cleanup attribute in the API module

**Review by lukstbit — APPROVED** [review]
LGTM

**Comment by lukstbit** [comment]
I merged #12250 if you want to rebase and get rid of that array of nullable strings.

**Review by BrayanDSO — APPROVED** [review]

**Pull request merged**